### PR TITLE
Simplify database operations in WebAPI::Controller::Test

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/Test.pm [hunk 1]
```diff
@@ -400,12 +388,8 @@ sub get_current_job {
...
     return $self->reply->not_found if (!defined $self->param('testid'));
 
-    my $job = $self->schema->resultset("Jobs")->search(
-        {
-            id => $self->param('testid')
-        },
-        {prefetch => qw(jobs_assets)})->first;
-    return $job;
+    return $self->schema->resultset("Jobs")->search({id => $self->param('testid')}, {prefetch => qw(jobs_assets)})
+      ->first;
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Controller/Test.pm:392]:
```suggestion
    return $self->schema->resultset('Jobs')->find($self->param('testid'}, {prefetch => qw(jobs_assets)});
```

**Review by Martchus — APPROVED** [review]

File: lib/OpenQA/WebAPI/Controller/Test.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/WebAPI/Controller/Test.pm]: should have tried a multi-line regex :D Fixed and found many more instances

**Comment by Martchus** [comment]
I will review this when https://github.com/os-autoinst/openQA/pull/4202 has been merged because I'm not checking each and every search/find substitution here again.

File: lib/OpenQA/WebAPI/Controller/Step.pm [hunk 2]
```diff
@@ -19,7 +31,7 @@ sub _init {
     my ($self) = @_;
 
     return 0 unless my $job = $self->app->schema->resultset('Jobs')->find($self->param('testid'));
-    my $module = $job->modules->search({name => $self->param('moduleid')})->first;
+    my $module = $job->modules->find({name => $self->param('moduleid')});
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Controller/Step.pm:34]: There's no unique constraint on the name column alone so this query might return multiple results. Better limit it to one row:

```suggestion
    my $module = $job->modules->find({name => $self->param('moduleid')}, {rows => 1});
```

It is a bit unfortunate that the database schema's unique constraints don't match the way API parameters are specified but I suppose improving that is out of scope for this PR. Just ensure we don't end up with Perl warnings in the logs due to this limitation.

---

Please also check whether other changes of that kind fall into the same category. A query via `find` must not return more than one row.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: lib/OpenQA/Schema/ResultSet/DeveloperSessions.pm [hunk 3]
```diff
@@ -14,34 +24,24 @@ use OpenQA::Schema::Result::DeveloperSessions;
 use OpenQA::WebSockets::Client;
 use OpenQA::Log 'log_error';
 
-sub register {
-    my ($self, $job_id, $user_id) = @_;
+sub _register_txn ($self, $job_id, $user_id) {
```
[the remark this answers is not in the document]
Comment (okurz) [reply lib/OpenQA/Schema/ResultSet/DeveloperSessions.pm]: used your suggestion, thxt

File: lib/OpenQA/WebAPI/Controller/Step.pm (hunk 2 above)
Comment (okurz) [reply lib/OpenQA/WebAPI/Controller/Step.pm]: I moved all "find" related changes to https://github.com/os-autoinst/openQA/pull/4202

**Pull request closed without merging**